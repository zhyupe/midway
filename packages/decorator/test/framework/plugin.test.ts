import {
  getClassMetadata,
  Plugin,
  INJECT_CUSTOM_TAG
} from '../../src';

class Test {
  @Plugin()
  test: any;

  @Plugin('bbb')
  bbb: any;
}

describe('/test/framework/plugin.test.ts', () => {
  it('plugin decorator should be ok', () => {
    let data = getClassMetadata(INJECT_CUSTOM_TAG, Test);
    expect(data).toMatchSnapshot();
  });
});
