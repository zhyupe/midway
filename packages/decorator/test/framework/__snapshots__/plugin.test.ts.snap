// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/framework/plugin.test.ts plugin decorator should be ok 1`] = `
Object {
  "bbb": Object {
    "key": "plugin",
    "propertyName": "bbb",
    "targetKey": "bbb",
  },
  "test": Object {
    "key": "plugin",
    "propertyName": "test",
    "targetKey": "test",
  },
}
`;
